chromadb/embedding_functions/__init__.py

```diff
@@ -0,0 +1,174 @@
+    OpenAIEmbeddingFunction,
+)
+from chromadb.embedding_functions.huggingface_embedding_function import (
+    HuggingFaceEmbeddingFunction,
+    HuggingFaceEmbeddingServer,
+)
+from chromadb.embedding_functions.sentence_transformer_embedding_function import (
+    SentenceTransformerEmbeddingFunction,
+)
+from chromadb.embedding_functions.google_embedding_function import (
+    GooglePalmEmbeddingFunction,
+    GoogleGenerativeAiEmbeddingFunction,
+    GoogleVertexEmbeddingFunction,
+)
+from chromadb.embedding_functions.ollama_embedding_function import (
+    OllamaEmbeddingFunction,
+)
+from chromadb.embedding_functions.instructor_embedding_function import (
+    InstructorEmbeddingFunction,
+)
+from chromadb.embedding_functions.jina_embedding_function import JinaEmbeddingFunction
+from chromadb.embedding_functions.voyageai_embedding_function import (
+    VoyageAIEmbeddingFunction,
+)
+from chromadb.embedding_functions.onnx_mini_lm_l6_v2 import ONNXMiniLM_L6_V2
+from chromadb.embedding_functions.open_clip_embedding_function import (
+    OpenCLIPEmbeddingFunction,
+)
+from chromadb.embedding_functions.roboflow_embedding_function import (
+    RoboflowEmbeddingFunction,
+)
+from chromadb.embedding_functions.text2vec_embedding_function import (
+    Text2VecEmbeddingFunction,
+)
+from chromadb.embedding_functions.amazon_bedrock_embedding_function import (
+    AmazonBedrockEmbeddingFunction,
+)
+from chromadb.embedding_functions.chroma_langchain_embedding_function import (
+    ChromaLangchainEmbeddingFunction,
+)
+
+# Dictionary of supported embedding functions
+supported_embedding_functions: Dict[str, Type[EmbeddingFunction[Documents]]] = {
+    "cohere": CohereEmbeddingFunction,
+    "openai": OpenAIEmbeddingFunction,
+    "huggingface": HuggingFaceEmbeddingFunction,
+    "huggingface_server": HuggingFaceEmbeddingServer,
+    "sentence_transformer": SentenceTransformerEmbeddingFunction,
+    "google_palm": GooglePalmEmbeddingFunction,
+    "google_generative_ai": GoogleGenerativeAiEmbeddingFunction,
+    "google_vertex": GoogleVertexEmbeddingFunction,
+    "ollama": OllamaEmbeddingFunction,
+    "instructor": InstructorEmbeddingFunction,
+    "jina": JinaEmbeddingFunction,
+    "voyageai": VoyageAIEmbeddingFunction,
+    "onnx_mini_lm_l6_v2": ONNXMiniLM_L6_V2,
+    "open_clip": OpenCLIPEmbeddingFunction,
+    "roboflow": RoboflowEmbeddingFunction,
+    "text2vec": Text2VecEmbeddingFunction,
+    "amazon_bedrock": AmazonBedrockEmbeddingFunction,
+    "chroma_langchain": ChromaLangchainEmbeddingFunction,
+}
+
+
+# Function to register custom embedding functions
+def register_embedding_function(ef_class: Type[EmbeddingFunction[Embeddable]]) -> None:
+    """Register a custom embedding function.
+
+    Args:
+        ef_class: The embedding function class to register.
+    """
+    # Create an instance to get the name
+    ef_instance = ef_class()
+    name = ef_instance.name()
```

Comment:
I don't think this is always safe, there could be EF constructors that have required arguments
could we make `.name()` a static method? I think that makes more sense semantically anyways